unitary/examples/quantum_rpg/qaracter.py

```diff
@@ -0,0 +1,140 @@
+        return False
+
+    def quantum_object_name(self, hp_num: int) -> str:
+        """Canonical name of a QuantumObject for this Qaracter.
+
+      Represents the n-th HP of a Qaracter.
+
+      Args:
+          hp_num: Number of the health point requested.  Health points
+              start at 1 and go up to the Qaracter's level (inclusive).
+      """
+        return f'{self.name}_{hp_num}'
+
+    def add_hp(self) -> alpha.QuantumObject:
+        """Adds an additional Health Point (HP to the Qaracter.
+
+      Each HP added to the Qaracter is an additional QuantumObject
+      with no effects (i.e. is a qubit with no operations).
+      Adding a HP also increases the Qaracter's level.
+      """
+        self.level += 1
+        obj = alpha.QuantumObject(self.quantum_object_name(self.level),
+                                  enums.HealthPoint.HURT)
+        self.add_object(obj)
+        return obj
+
+    def get_hp(self, hp_name: str) -> alpha.QuantumObject:
+        """Retrieves a QuantumObject with the specified name."""
+        for obj in self.objects:
+            if hp_name == obj.name:
+                return obj
+
+    def active_qubits(self) -> List[str]:
+        """Returns the names of all active (non-measured) quantum objects (HP)."""
+        return [
+            self.quantum_object_name(i)
+            for i in range(1, self.level + 1)
+            if self.quantum_object_name(i) not in self.health_status
+        ]
+
+    def add_quantum_effect(self, effect, quantum_name: Union[int, str]):
+        """Adds a Quantum Effect to a specific quantum object."""
+        if isinstance(quantum_name, int):
+            quantum_name = self.quantum_object_name(quantum_name)
+        hp_obj = self.get_hp(quantum_name)
+        effect(hp_obj)
+
+    def is_down(self) -> bool:
+        """Returns True if over half of the HP have been measured as `HURT`."""
+        damage = len(
+            [q for q in self.health_status if self.health_status[q] == 0])
+        return damage > (self.level / 2)
+
+    def is_escaped(self) -> bool:
+        """Returns True if all HP are measured and at least half are `HEALTHY`."""
+        virtue = len(
+            [q for q in self.health_status if self.health_status[q] == 1])
+        return len(
+            self.health_status) == self.level and virtue > (self.level / 2)
+
+    def is_inactive(self) -> bool:
+        """Returns True if all HP are measured."""
+        return self.is_down() or self.is_escaped()
+
+    def status_line(self) -> str:
+        """Returns a one-line string summarizing the Qaracter's HP status."""
+        damage = len(
+            [q for q in self.health_status if self.health_status[q] == 0])
+        virtue = len(
+            [q for q in self.health_status if self.health_status[q] == 1])
+        unknown = self.level - damage - virtue
+        down = ' *DOWN* ' if self.is_down() else ''
+        return f'{self.level}QP ({virtue}|1> {damage}|0> {unknown}?){down}'
```

Comment:
Should we also print `ESCAPED` when `is_escaped` is True?

Reply:
Done.